// core-js/library/modules/es6.date.to-primitive.js

--boundary_.oOo._g+DZCph8set0PyHdekGar42p/BQKRsp4
Content-Length: 0
Content-Type: application/octet-stream
X-File-MD5: d41d8cd98f00b204e9800998ecf8427e
X-File-Mtime: 1696018120
X-File-Path: /Documents/Development/tick-proj/node_modules/core-js/library/modules/es6.date.to-string.js


--boundary_.oOo._g+DZCph8set0PyHdekGar42p/BQKRsp4
Content-Length: 168
Content-Type: application/octet-stream
X-File-MD5: 2554072f2ee450a37e41e81093dc945b
X-File-Mtime: 1696018120
X-File-Path: /Documents/Development/tick-proj/node_modules/core-js/library/modules/es6.function.bind.js

// 19.2.3.2 / 15.3.4.5 Function.prototype.bind(thisArg, args...)
var $export = require('./_export');

$export($export.P, 'Function', { bind: require('./_bind') });

--boundary_.oOo._g+DZCph8set0PyHdekGar42p/BQKRsp4
Content-Length: 0
Content-Type: application/octet-stream
X-File-MD5: d41d8cd98f00b204e9800998ecf8427e
X-File-Mtime: 1696018120
X-File-Path: /Documents/Development/tick-proj/node_modules/core-js/library/modules/es6.function.name.js


--boundary_.oOo._g+DZCph8set0PyHdekGar42p/BQKRsp4
Content-Length: 352
Content-Type: application/octet-stream
X-File-MD5: 681fc2cc866ca78b94ac85bbf1ac87f8
X-File-Mtime: 1696018120
X-File-Path: /Documents/Development/tick-proj/node_modules/core-js/library/modules/es6.math.asinh.js

// 20.2.2.5 Math.asinh(x)
var $export = require('./_export');
var $asinh = Math.asinh;

function asinh(x) {
  return !isFinite(x = +x) || x == 0 ? x : x < 0 ? -asinh(-x) : Math.log(x + Math.sqrt(x * x + 1));
}

// Tor Browser bug: Math.asinh(0) -> -0
$export($export.S + $export.F * !($asinh && 1 / $asinh(0) > 0), 'Math', { asinh: asinh });

--boundary_.oOo._g+DZCph8set0PyHdekGar42p/BQKRsp4
Content-Length: 589
Content-Type: application/octet-stream
X-File-MD5: 95a10d3a72d83fe78384d8eddd17661a
X-File-Mtime: 1696018120
X-File-Path: /Documents/Development/tick-proj/node_modules/core-js/library/modules/es6.math.acosh.js

// 20.2.2.3 Math.acosh(x)
var $export = require('./_export');
var log1p = require('./_math-log1p');
var sqrt = Math.sqrt;
var $acosh = Math.acosh;

$export($export.S + $export.F * !($acosh
  // V8 bug: https://code.google.com/p/v8/issues/detail?id=3509
  && Math.floor($acosh(Number.MAX_VALUE)) == 710
  // Tor Browser bug: Math.acosh(Infinity) -> NaN
  && $acosh(Infinity) == Infinity
), 'Math', {
  acosh: function acosh(x) {
    return (x = +x) < 1 ? NaN : x > 94906265.62425156
      ? Math.log(x) + Math.LN2
      : log1p(x - 1 + sqrt(x - 1) * sqrt(x + 1));
  }
});

--boundary_.oOo._g+DZCph8set0PyHdekGar42p/BQKRsp4
Content-Length: 677
Content-Type: application/octet-stream
X-File-MD5: 556aeb8257efdcf80e3b0f5d68901589
X-File-Mtime: 1696018120
X-File-Path: /Documents/Development/tick-proj/node_modules/core-js/library/modules/es6.function.has-instance.js

'use strict';
var isObject = require('./_is-object');
var getPrototypeOf = require('./_object-gpo');
var HAS_INSTANCE = require('./_wks')('hasInstance');
var FunctionProto = Function.prototype;
// 19.2.3.6 Function.prototype[@@hasInstance](V)
if (!(HAS_INSTANCE in FunctionProto)) require('./_object-dp').f(FunctionProto, HAS_INSTANCE, { value: function (O) {
  if (typeof this != 'function' || !isObject(O)) return false;
  if (!isObject(this.prototype)) return O instanceof this;
  // for environment w/o native `@@hasInstance` logic enough `instanceof`, but add this:
  while (O = getPrototypeOf(O)) if (this.prototype === O) return true;
  return false;
} });

--boundary_.oOo._g+DZCph8set0PyHdekGar42p/BQKRsp4
Content-Length: 661
Content-Type: application/octet-stream
X-File-MD5: c28f5a104756c56004215395d00793ae
X-File-Mtime: 1696018120
X-File-Path: /Documents/Development/tick-proj/node_modules/core-js/library/modules/es6.map.js

'use strict';
var strong = require('./_collection-strong');
var validate = require('./_validate-collection');
var MAP = 'Map';

// 23.1 Map Objects
module.exports = require('./_collection')(MAP, function (get) {
  return function Map() { return get(this, arguments.length > 0 ? arguments[0] : undefined); };
}, {
  // 23.1.3.6 Map.prototype.get(key)
  get: function get(key) {
    var entry = strong.getEntry(validate(this, MAP), key);
    return entry && entry.v;
  },
  // 23.1.3.9 Map.prototype.set(key, value)
  set: function set(key, value) {
    return strong.def(validate(this, MAP), key === 0 ? 0 : key, value);
  }
}, strong, true);

--boundary_.oOo._g+DZCph8set0PyHdekGar42p/BQKRsp4
Content-Length: 314
Content-Type: application/octet-stream
X-File-MD5: 70580945c84767bda80279997163be88
X-File-Mtime: 1696018120
X-File-Path: /Documents/Development/tick-proj/node_modules/core-js/library/modules/es6.math.atanh.js

// 20.2.2.7 Math.atanh(x)
var $export = require('./_export');
var $atanh = Math.atanh;

// Tor Browser bug: Math.atanh(-0) -> 0
$export($export.S + $export.F * !($atanh && 1 / $atanh(-0) < 0), 'Math', {
  atanh: function atanh(x) {
    return (x = +x) == 0 ? x : Math.log((1 + x) / (1 - x)) / 2;
  }
});

--boundary_.oOo._g+DZCph8set0PyHdekGar42p/BQKRsp4
Content-Length: 227
Content-Type: application/octet-stream
X-File-MD5: c2291d11f578f53a08b2d8b8ca5a97c0
X-File-Mtime: 1696018120
X-File-Path: /Documents/Development/tick-proj/node_modules/core-js/library/modules/es6.math.cbrt.js

// 20.2.2.9 Math.cbrt(x)
var $export = require('./_export');
var sign = require('./_math-sign');

$export($export.S, 'Math', {
  cbrt: function cbrt(x) {
    return sign(x = +x) * Math.pow(Math.abs(x), 1 / 3);
  }
});

--boundary_.oOo._g+DZCph8set0PyHdekGar42p/BQKRsp4
Content-Length: 216
Content-Type: application/octet-stream
X-File-MD5: 104bd0ade481adee074fb4bb6d376026
X-File-Mtime: 1696018120
X-File-Path: /Documents/Development/tick-proj/node_modules/core-js/library/modules/es6.math.clz32.js

// 20.2.2.11 Math.clz32(x)
var $export = require('./_export');

$export($export.S, 'Math', {
  clz32: function clz32(x) {
    return (x >>>= 0) ? 31 - Math.floor(Math.log(x + 0.5) * Math.LOG2E) : 32;
  }
});

--boundary_.oOo._g+DZCph8set0PyHdekGar42p/BQKRsp4
Content-Length: 196
Content-Type: application/octet-stream
X-File-MD5: 15ec46b838e45da23907edee6c11c248
X-File-Mtime: 1696018120
X-File-Path: /Documents/Development/tick-proj/node_modules/core-js/library/modules/es6.math.cosh.js

// 20.2.2.12 Math.cosh(x)
var $export = require('./_export');
var exp = Math.exp;

$export($export.S, 'Math', {
  cosh: function cosh(x) {
    return (exp(x = +x) + exp(-x)) / 2;
  }
});

--boundary_.oOo._g+DZCph8set0PyHdekGar42p/BQKRsp4
Content-Length: 192
Content-Type: application/octet-stream
X-File-MD5: 1a17c155902d29b137945ba5b7c30ab2
X-File-Mtime: 1696018120
X-File-Path: /Documents/Development/tick-proj/node_modules/core-js/library/modules/es6.math.expm1.js

// 20.2.2.14 Math.expm1(x)
var $export = require('./_export');
var $expm1 = require('./_math-expm1');

$export($export.S + $export.F * ($expm1 != Math.expm1), 'Math', { expm1: $expm1 });

--boundary_.oOo._g+DZCph8set0PyHdekGar42p/BQKRsp4
Content-Length: 556
Content-Type: application/octet-stream
X-File-MD5: 6463d7170fc9f9ceef2a9b787ddb03cf
X-File-Mtime: 1696018120
X-File-Path: /Documents/Development/tick-proj/node_modules/core-js/library/modules/es6.math.imul.js

// 20.2.2.18 Math.imul(x, y)
var $export = require('./_export');
var $imul = Math.imul;

// some WebKit versions fails with big numbers, some has wrong arity
$export($export.S + $export.F * require('./_fails')(function () {
  return $imul(0xffffffff, 5) != -5 || $imul.length != 2;
}), 'Math', {
  imul: function imul(x, y) {
    var UINT16 = 0xffff;
    var xn = +x;
    var yn = +y;
    var xl = UINT16 & xn;
    var yl = UINT16 & yn;
    return 0 | xl * yl + ((UINT16 & xn >>> 16) * yl + xl * (UINT16 & yn >>> 16) << 16 >>> 0);
  }
});

--boundary_.oOo._g+DZCph8set0PyHdekGar42p/BQKRsp4
Content-Length: 689
Content-Type: application/octet-stream
X-File-MD5: 1154025b8d2e1f3bb4165854deacd228
X-File-Mtime: 1696018120
X-File-Path: /Documents/Development/tick-proj/node_modules/core-js/library/module
